### PR TITLE
boilr: init at 1.9.4

### PR DESCRIPTION
**Pull request by foolnotion**
## Description of changes

BoilR is a utility which automatically adds games to the Steam library. This PR adds boiler for `x86_64-linux`. It may be possible to run it on other platforms but I have no way of testing that.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by eclairevoyant — DISMISSED** [review]

File: pkgs/games/boilr/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  '';
+
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description =
+      "BoilR automatically adds (almost) all your games to your Steam library (including image art)";
+    homepage = "https://github.com/PhilipK/BoilR";
+    licenses = with lib.licenses; [ asl20 mit];
+    platforms = platforms.all;
```
Comment (eclairevoyant) [inline pkgs/games/boilr/default.nix:49]: specify explicitly supported platforms. I doubt this is reall buildable on all (https://github.com/NixOS/nixpkgs/blob/master/lib/systems/doubles.nix)

Comment (foolnotion) [reply pkgs/games/boilr/default.nix:49]: would it be better to just not specify the platform? according to the manual:

> If no platform is specified, the package does not have prebuilt binaries

Comment (eclairevoyant) [reply pkgs/games/boilr/default.nix:49]: no, then hydra won't cache it at all

**Comment by foolnotion** [comment]
Hi, is there anything else required from my side to complete this PR? Thanks!

**Comment by MalteT** [comment]
Result of `nixpkgs-review pr 247991` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>boilr</li>
  </ul>
</details>

**Comment by MalteT** [comment]
I'm using this package definition for a while now, runs without any issues for me on wayland. LGTM

**Review by stefan-bordei — APPROVED** [review]
LGTM

**Review by iliana — APPROVED** [review]
LGTM and runs on my machine; I would suggest installing the .desktop file and its icon:

File: pkgs/games/boilr/default.nix [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  cargoHash = "sha256-5nhtGqukZ8tp7gPV+JiUEHTzWUj6JtsB2i7bODcXJSc=";
+
+  nativeBuildInputs = [ perl ];
+
+  buildInputs = rpathLibs;
+
+  postInstall = ''
+    patchelf --add-rpath "${lib.makeLibraryPath rpathLibs}" $out/bin/boilr
+  '';
```
[the remark this answers is not in the document]
Comment (foolnotion) [reply pkgs/games/boilr/default.nix]: done!

**Review by stefan-bordei — APPROVED** [review]

File: pkgs/by-name/bo/boilr/package.nix [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    patchelf --add-rpath "${lib.makeLibraryPath rpathLibs}" $out/bin/boilr
+    install -Dpm 0644 flatpak/io.github.philipk.boilr.desktop $out/share/applications/boilr.desktop
+    install -Dpm 0644 resources/io.github.philipk.boilr.png $out/share/icons/io.github.philipk.boilr.png
+  '';
+
+  dontPatchELF = true;
+
+  meta = {
+    description =
+      "BoilR automatically adds (almost) all your games to your Steam library (including image art)";
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/boilr/package.nix:61]:
```suggestion
      "Automatically adds (almost) all your games to your Steam library (including image art)";
```

File: pkgs/by-name/bo/boilr/package.nix [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    homepage = "https://github.com/PhilipK/BoilR";
+    license = with lib.licenses; [ asl20 mit ];
+    platforms = lib.platforms.linux;
+    maintainers = with lib.maintainers; [ foolnotion ];
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/boilr/package.nix:65]:
```suggestion
    maintainers = with lib.maintainers; [ foolnotion ];
    mainProgram = "boilr";
```

File: pkgs/by-name/bo/boilr/package.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  cargoHash = "sha256-nAZU1xVpeRXubotla4I6InGMH4lisPMOnoqaK5mBPCM=";
+
+  nativeBuildInputs = [ perl ];
+
+  buildInputs = rpathLibs;
+
+  postInstall = ''
+    patchelf --add-rpath "${lib.makeLibraryPath rpathLibs}" $out/bin/boilr
+    install -Dpm 0644 flatpak/io.github.philipk.boilr.desktop $out/share/applications/boilr.desktop
+    install -Dpm 0644 resources/io.github.philipk.boilr.png $out/share/icons/io.github.philipk.boilr.png
```
Comment (Aleksanaa) [inline pkgs/by-name/bo/boilr/package.nix:54]:
```suggestion
    install -Dpm 0644 resources/io.github.philipk.boilr.png $out/share/pixmaps/io.github.philipk.boilr.png
```

icons are usually for organized icon themes. You can also install them to `icons/hicolor/<>x<>/apps/`. Also see https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html

**Review by Aleksanaa — COMMENTED** [review]
Sorry for the delay.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**